test/core/index.test.js

```diff
@@ -322,6 +322,45 @@ describe('Core Tests', () => {
     assert.strictEqual(data, body);
   });
 
+  it('supports text multibyte body', async () => {
+    const method = 'POST';
+    const body = 'こんにちは';
+    const resp = await defaultCtx.request('https://httpbin.org/post', { method, body });
+    assert.strictEqual(resp.statusCode, 200);
+    assert.strictEqual(resp.headers['content-type'], 'application/json');
+    const buf = await readStream(resp.readable);
+    const jsonResponseBody = JSON.parse(buf);
+    assert(typeof jsonResponseBody === 'object');
+    const { data, headers } = jsonResponseBody;
+    assert.strictEqual(headers['Content-Type'], 'text/plain; charset=utf-8');
+    assert.strictEqual(+headers['Content-Length'], Buffer.from(body, 'utf-8').length);
+    assert.strictEqual(data, body);
+  });
+
+  it('supports buffer body', async () => {
+    const method = 'POST';
+    const body = Buffer.from(new Uint8Array([0xfe, 0xff, 0x41]));
```

Comment:
I tried to create a buffer, where `Buffer.byteLength(opts.body, 'utf-8');` would return a wrong length, but I didn't succeed :-)